session/schema_amender.go

```diff
@@ -426,7 +423,16 @@ func (a *amendOperationAddNewIndex) processRowKey(ctx context.Context, sctx sess
 	if err != nil {
 		return errors.Trace(err)
 	}
-	resAddMutations.Push(pb.Op_Put, newIdxKey, newIdxValue, false)
+	newIndexOp := pb.Op_Put
+	if a.info.indexInfoAtCommit.Meta().Unique {
+		if _, ok := a.processedNewIndexKeys[string(newIdxKey)]; ok {
```

Comment:
Only need to check current-txn's keys?

Reply:
The new mutations generated will use `Op_insert` as its `Op_type`, the existence check will be done in the amend prewrite phase like insert statements in optimistic transactions.

Reply:
ok

Reply:
The existency check of the newly generated unique index keys will be done in the pessimistic lock phase during amend, like the normal path. Except that there is no pessmistic write conflict retry.